[Offline DQM] Fixes for calo layer 2

kreczko opened the pull request:
- fixes overflow bug
 - adds control plots for jets & sums
 - improves titles for the denominator and numerator histograms

@rekovic @dmitrijus please check & sign

kreczko committed:
fixing overflow bin bug

kreczko committed:
improving title for efficiency histograms

kreczko committed:
added control plots for jets & sums (calo layer 2)

cmsbuild commented:
A new Pull Request was created by  @kreczko (Luke Kreczko) for master.

It involves the following packages:

DQMOffline/L1Trigger


@dmitrijus, @cmsbuild, @rekovic, @vanbesien, @mulhearn, @davidlange6 can you please review it and eventually sign? Thanks.
@thomreis, @rociovilar this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 9f7436f38f0e5ccb904441d405af969ced935235
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18863/20064/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18863/20064/summary.html

Comparison Summary:
 - You potentially added 181 lines to the logs
 - Reco comparison results: 3357 differences found in the comparisons
 - DQMHistoTests: Total files compared: 24
 - DQMHistoTests: Total histograms compared: 1832990
 - DQMHistoTests: Total failures: 62832
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1769978
 - DQMHistoTests: Total skipped: 180
 - DQMHistoTests: Total Missing objects: 0
 - Checked 98 log files, 14 edm output root files, 24 DQM output files

davidlange6 commented on DQMOffline/L1Trigger/interface/L1TStage2CaloLayer2Offline.h:
```diff
@@ -111,6 +111,9 @@ class L1TStage2CaloLayer2Offline: public DQMEDAnalyzer {
   MonitorElement* h_nVertex_;
 
+  // energy sums control plots
+  std::map<std::string, MonitorElement*> h_energySum_controlPlots_;
```
Hi @kreczko - please change the std::string here and below into a set of enums to avoid endless string comparisons in the event loop.

kreczko committed:
changing map<string,MonitorElement*> to map<Enum,MonitorElement*>

cmsbuild commented:
Pull request #18863 was updated. @dmitrijus, @cmsbuild, @rekovic, @vanbesien, @mulhearn, @davidlange6 can you please check and sign again.

kreczko commented:
Hi @davidlange6, thanks for the feedback.
Added `L1TStage2CaloLayer2Offline::ControlPlots` enum and modified the map.

@dmitrijus & @rekovic please check and sign again.

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: ae2663f71c1006742a4393320c1b1014044dbefd
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18863/20136/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18863/20136/summary.html

Comparison Summary:
 - You potentially added 192 lines to the logs
 - Reco comparison results: 3394 differences found in the comparisons
 - DQMHistoTests: Total files compared: 24
 - DQMHistoTests: Total histograms compared: 1833626
 - DQMHistoTests: Total failures: 48904
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1784542
 - DQMHistoTests: Total skipped: 180
 - DQMHistoTests: Total Missing objects: 0
 - Checked 98 log files, 14 edm output root files, 24 DQM output files

kreczko commented:
@rekovic @mulhearn please review & sign for L1T

kreczko commented:
@rekovic @mulhearn please review & sign for L1T

rekovic commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

davidlange6 commented:
+1
